### PR TITLE
Making 'api.queue.Queue' inherit from 'api.ext.Topic'

### PR DESCRIPTION
**Pull request by PierreSavatte**
Also adding a retry decorator to handle Kafka errors.
Can also execute tests with a running Kafka instance if a env var named KAFKA_BOOTSTRAP_SERVERS is found.

### CODE REVIEW

File: kirby/api/ext.py [hunk 1]
```diff
@@ -1,61 +1,145 @@
...
 import msgpack
 from kafka import KafkaConsumer, KafkaProducer
+from kafka.errors import NoBrokersAvailable
+
+logger = logging.getLogger(__name__)
+
+
+def retry(function):
+
+    NB_RETRY = 3
```
Comment (ericgazoni) [inline kirby/api/ext.py]: Why `3`, and why not getting it from the environment? 

Can you move it to the module level please?

File: kirby/api/ext.py [hunk 2]
```diff
@@ -1,61 +1,145 @@
...
+
+    def function_decorated():
+        for counter in range(NB_RETRY):
+            try:
+                return_value = function_decorated()
+            except NoBrokersAvailable as e:
+                if counter == NB_RETRY:
```
Comment (ericgazoni) [inline kirby/api/ext.py]: watchdog checks should always be done using `>=` comparisons, to ensure you don't end up in an endless loop

File: kirby/api/ext.py [hunk 3]
```diff
@@ -1,61 +1,145 @@
+import logging
+import datetime
 import msgpack
 from kafka import KafkaConsumer, KafkaProducer
+from kafka.errors import NoBrokersAvailable
+
+logger = logging.getLogger(__name__)
+
+
+def retry(function):
```
Comment (ericgazoni) [inline kirby/api/ext.py]: why implementing your own retry [when](https://github.com/jd/tenacity) [alternatives](https://github.com/invl/retry) [exist](https://github.com/rholder/retrying) ?

File: kirby/api/queue.py [hunk 4]
```diff
@@ -3,13 +3,13 @@
...
-import datetime
+from .ext import Topic
 
 logger = logging.getLogger(__name__)
 
 
-class Queue:
-    def __init__(self, name, testing=False):
+class Queue(Topic):
+    def __init__(self, name, testing=False, security_protocol="SSL"):
```
Comment (ericgazoni) [inline kirby/api/queue.py]: is adding the security protocol really needed here?

File: tests/conftest.py [hunk 5]
```diff
@@ -250,26 +250,47 @@ def kafka_topic_factory():
...
-        admin.create_topics([NewTopic(topic_name, 1, 1)], timeout_ms=1500)
-        yield
+        return create_kafka_topic
 
-        admin.delete_topics([topic_name])
-        admin.close()
+    else:
+        logger.warning(
+            f"There is no KAFKA_BOOTSTRAP_SERVERS. "
+            f"Creation of kafka_topic passed."
```
Comment (ericgazoni) [inline tests/conftest.py]: you probably want to use "skipped" instead of "passed" here

File: kirby/api/ext.py (hunk 1 above)
Comment (PierreSavatte) [reply kirby/api/ext.py]: Now get from the environment variables.

File: kirby/api/ext.py (hunk 2 above)
Comment (PierreSavatte) [reply kirby/api/ext.py]: Not used anymore

File: kirby/api/ext.py (hunk 3 above)
Comment (PierreSavatte) [reply kirby/api/ext.py]: Using tenacity now

File: kirby/api/queue.py (hunk 4 above)
Comment (PierreSavatte) [reply kirby/api/queue.py]: It is now a boolean, if ssl_security_protocol is false, use plaintext

File: tests/conftest.py (hunk 5 above)
Comment (PierreSavatte) [reply tests/conftest.py]: Corrected.

**Pull request merged**